wdl/wdl-core/src/main/scala/wdl/Scatter.scala

```diff
@@ -39,7 +40,10 @@ object Scatter {
   /**
     * @param preserveIndexForOuterLookups When we're evaluating the scatter collection, should we preserve scatter index when we have to use the outerLookup?
     */
-  def womScatterNode(scatter: Scatter, localLookup: Map[String, GraphNodePort.OutputPort], outerLookup: Map[String, OutputPort], preserveIndexForOuterLookups: Boolean): ErrorOr[ScatterNodeWithNewNodes] = {
+  def womScatterNode(scatter: Scatter,
+                     localLookup: Map[String, GraphNodePort.OutputPort],
+                     outerLookup: Map[String, OutputPort],
+                     preserveIndexForOuterLookups: Boolean)(implicit wdlVersionSpecifics: WdlVersionSpecifics): ErrorOr[ScatterNodeWithNewNodes] = {
```

Comment:
Since almost everything has expressions, and expressions are one of the things I tried to split, it means that almost everything now has a `(implicit wdlVersionSpecifics: WdlVersionSpecifics)` on the end of its constructor

Reply:
If it wasn't for the `implicit` I think all this would basically be a reader monad. 😮